foreshadow/steps/cleaner.py

```diff
@@ -69,33 +114,22 @@ def transform(self, X, *args, **kwargs):
         Returns:
             A transformed dataframe.
 
-        """
-        Xt = super().transform(X, *args, **kwargs)
-        self._check_empty_dataframe(Xt)
-        return Xt.dropna(axis=1, how="all")
+        Raises:
+            ValueError: new empty columns detected.
 
-    def _check_empty_dataframe(self, X) -> NoReturn:
-        """Check if all columns are empty in the dataframe.
+        """
+        if self.empty_columns is None:
```

Comment:
Please look at this logic again as we discussed. Consider this function will be called in both training and prediction and the decision we made not to abort if there are new empty columns during prediction.

Reply:
After adding several breakpoints and rerun the test, it turned out that our `fit_transform` method never calls the transform method. This seems a bit odd based on the convention that `fit_transform` calls `fit` and then `transform`. 

I'll look a bit deeper on this but as for now, it means we have to have this duplicate logic of checking empty columns in both `fit_transform` and `transform`.

Reply:
During training, we call `shadow.fit` and it should never trigger `transform`. So why are we overriding the `fit_transform` method?

It turns out that `cleaner` is a `preparestep`, whose `fit` method calls its `fit_transform` method. This seems to be an issue we have identified before and have stayed in the backlog for a while. 

The fix may not be straightforward since the current implementation involves multiple classes back to invocation in Sklean's `pipeline.py`.

I would recommend that we contain the change in the cleaner class and plan the work on the larger issue with the version upgrade task.